samza-sql/src/main/java/org/apache/samza/sql/translator/ScanTranslator.java

```diff
@@ -125,9 +126,11 @@ public void init(Context context) {
     }
 
     @Override
-    public SamzaSqlRelMessage apply(KV<Object, Object> message) {
+    public SamzaSqlRelMessage apply(SamzaSqlMessage samzaSqlMessage) {
       Instant startProcessing = Instant.now();
-      SamzaSqlRelMessage retMsg = this.msgConverter.convertToRelMessage(message);
+      SamzaSqlRelMessage retMsg = this.msgConverter.convertToRelMessage(samzaSqlMessage.keyAndMessageKV);
+      retMsg.getSamzaSqlRelMsgMetadata().setEventTime(samzaSqlMessage.metadata.getEventTime());
```

Comment:
This is an anti-pattern. 

1. When you are doing a get, You should expect that the object that is being returned is immutable. 
2. And in the implementation of get it is always a good idea to not return a mutable object. 

Ideally you should be able to initialize the SamzaSqlRelMessage with the metadata in the conversion layer that converts the SamzaSqlMessage to SamzaSqlRelMessage. It looks like you did this to not change the RelConverter interface. Can you add a comment saying that this should be fixed? RelConverter interface needs a cleanup, We can cleanup this code as part of that work.

Reply:
created SAMZA-2089 to track this & added comment